Automattic/wp-calypso
# Invites: Remove InviteHeader component

#### Changes proposed in this Pull Request

* The `InviteHeader` component is redundant; it repeats information already provided in the invite email and the following form. 
* This PR removes the component, bringing the primary actions up in the page hierarchy and simplifying the invite page.

**Before**

<img width="1280" alt="Screen Shot 2019-04-22 at 12 18 12 PM" src="https://user-images.githubusercontent.com/2124984/56511275-38429500-64fa-11e9-820c-475f95e23485.png">

**After**

<img width="1280" alt="Screen Shot 2019-04-22 at 12 17 58 PM" src="https://user-images.githubusercontent.com/2124984/56511273-3547a480-64fa-11e9-8025-befd363d310c.png">

#### Testing instructions

* Switch to this PR and ensure you have two different user accounts
* Invite one user to manage a site on the other user's account
* Click the Accept Invite link in the email
* Note the changes in the invite acceptance page; you should no longer site the site block above the confirmation form.

Fixes #32394
